hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -342,6 +353,35 @@ public SingleTransactionRecordBuilderImpl transactionID(@NonNull final Transacti
         return this;
     }
 
+    /**
+     * When we update nonce on the record, we need to update the body as well with the same transactionID.
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl syncBodyIdFromRecordId() {
+        final var newTransactionID = transactionID;
+        try {
+            final var signedTransaction = SignedTransaction.PROTOBUF.parseStrict(
+                    transaction.signedTransactionBytes().toReadableSequentialData());
+            final var existingTransactionBody =
+                    TransactionBody.PROTOBUF.parse(signedTransaction.bodyBytes().toReadableSequentialData());
+            final var body = existingTransactionBody
+                    .copyBuilder()
+                    .transactionID(newTransactionID)
+                    .build();
+            final var newBodyBytes = TransactionBody.PROTOBUF.toBytes(body);
+            final var newSignedTransaction =
+                    SignedTransaction.newBuilder().bodyBytes(newBodyBytes).build();
+            final var signedTransactionBytes = SignedTransaction.PROTOBUF.toBytes(newSignedTransaction);
+            this.transaction = Transaction.newBuilder()
+                    .signedTransactionBytes(signedTransactionBytes)
+                    .build();
+            return this;
+        } catch (Exception e) {
+            return this;
```

Comment:
Ofc it's all-but-impossible to reach this `catch` clause; but if we _did_, we'd prob want to propagate `e` via  `throw new IllegalStateException(e)` or the like.

Reply:
Threw RunTimeException